app/models/result.rb

```diff
@@ -120,4 +121,16 @@ def check_for_nil_marks
     end
     true
   end
+
+  def create_marks
+    @assignment = self.submission.assignment
```

Comment:
Don't use `@` here, all these variables should be local, not instance variables.